app/services/find_or_register_user.rb

```diff
@@ -0,0 +1,27 @@
+class FindOrRegisterUser
+  attr_reader :github_id
+
+  def initialize(params)
+    @github_id = params[:github_id]
+  end
+
+  def perform
+    user = found_user = find_user
+
+    if found_user.nil?
+      user = register_user
```

Comment:
we would never hit this case right now because we create the user at https://github.com/18F/micropurchase/blob/0f1958ca89b8778516b6b105f904cd9ddf5b6f77/app/models/user.rb#L14 -- let's bring that logic into here rather than creating and then registering